components/Cards/src/MDCCardCollectionCell.m

```diff
@@ -0,0 +1,340 @@
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _shadowElevations = [coder decodeObjectForKey:MDCCardCellShadowElevationsKey];
+    _shadowColors = [coder decodeObjectForKey:MDCCardCellShadowColorsKey];
+    _borderWidths = [coder decodeObjectForKey:MDCCardCellBorderWidthsKey];
+    _borderColors = [coder decodeObjectForKey:MDCCardCellBorderColorsKey];
+    _inkView = [coder decodeObjectForKey:MDCCardCellInkViewKey];
+    _selectedImageView = [coder decodeObjectForKey:MDCCardCellSelectedImageViewKey];
+    _state = [coder decodeIntegerForKey:MDCCardCellStateKey];
+    _selectable = [coder decodeBoolForKey:MDCCardCellSelectableKey];
+    [self commonMDCCardCollectionCellInit];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeObject:_shadowElevations forKey:MDCCardCellShadowElevationsKey];
+  [coder encodeObject:_shadowColors forKey:MDCCardCellShadowColorsKey];
+  [coder encodeObject:_borderWidths forKey:MDCCardCellBorderWidthsKey];
+  [coder encodeObject:_borderColors forKey:MDCCardCellBorderColorsKey];
+  [coder encodeObject:_inkView forKey:MDCCardCellInkViewKey];
+  [coder encodeObject:_selectedImageView forKey:MDCCardCellSelectedImageViewKey];
+  [coder encodeInteger:_state forKey:MDCCardCellStateKey];
+  [coder encodeBool:_selectable forKey:MDCCardCellSelectableKey];
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCCardCollectionCellInit];
+  }
+  return self;
+}
+
+- (void)commonMDCCardCollectionCellInit {
+  if (_inkView == nil) {
+    _inkView = [[MDCInkView alloc] initWithFrame:self.bounds];
+    _inkView.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+    _inkView.usesLegacyInkRipple = NO;
+    _inkView.layer.zPosition = FLT_MAX;
+    [self addSubview:self.inkView];
+  }
+
+  [self initializeSelectedImage];
+
+  if (_shadowElevations == nil) {
+    _shadowElevations = [NSMutableDictionary dictionary];
+    _shadowElevations[@(MDCCardCellStateNormal)] = @(1.f);
+    _shadowElevations[@(MDCCardCellStateHighlighted)] = @(8.f);
+    _shadowElevations[@(MDCCardCellStateSelected)] = @(8.f);
+  }
+
+  if (_shadowColors == nil) {
+    _shadowColors = [NSMutableDictionary dictionary];
+    _shadowColors[@(MDCCardCellStateNormal)] = [UIColor blackColor];
+  }
+
+  if (_borderColors == nil) {
+    _borderColors = [NSMutableDictionary dictionary];
+  }
+
+  if (_borderWidths == nil) {
+    _borderWidths = [NSMutableDictionary dictionary];
+    _borderWidths[@(MDCCardCellStateNormal)] = @(0.f);
+  }
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  [self updateShadowElevation];
+  [self updateShadowColor];
+  [self updateBorderWidth];
+
+  self.cornerRadius = 4.f;
+  self.layer.shadowPath = [self boundingPath].CGPath;
+}
+
+- (void)initializeSelectedImage {
```

Comment:
This is only used once, consider leaving it inline?
(I know some people are opposed to long functions, but I consider init* an exception.

Reply:
Let's be very careful here. The fact that the return type is `void` keeps it out of the `init` method family.  We may want to consider renaming down the line or adding an inline comment.

http://clang.llvm.org/docs/AutomaticReferenceCounting.html#method-families